core/src/main/java/io/kestra/core/tasks/storages/Filter.java

```diff
@@ -0,0 +1,221 @@
+                } catch (IllegalVariableEvaluationException e) {
+                    exception = e;
+                }
+
+                FilterType action = this.filterType;
+
+                if (match == null) {
+                    switch (errorOrNullBehavior) {
+                        case FAIL -> {
+                            if (exception != null) {
+                                throw exception;
+                            } else {
+                                throw new IllegalVariableEvaluationException(String.format(
+                                    "Expression `%s` return `null` on record `%s`",
+                                    filterCondition,
+                                    line
+                                ));
+                            }
+                        }
+                        case INCLUDE -> action = FilterType.INCLUDE;
+                        case EXCLUDE ->  action = FilterType.EXCLUDE;
+                    }
+                    match = true;
+                }
+
+                if (!match) {
+                    action = action.reverse();
+                }
+
+                switch (action) {
+                    case INCLUDE -> {
+                        writer.write(line);
+                        writer.newLine();
+                    }
+                    case EXCLUDE -> droppedRecordTotal++;
+                }
+
+            }
+        }
+        URI uri = runContext.storage().putFile(path.toFile());
+        return Output.builder().uri(uri).droppedRecordsTotal(droppedRecordTotal).build();
+    }
+
+    private PebbleExpressionPredicate getExpressionPredication(RunContext runContext) {
+        return new PebbleExpressionPredicate(runContext, filterCondition);
+    }
+
+    private BufferedReader newBufferedReader(final RunContext runContext, final URI objectURI) throws IOException {
+        InputStream is = runContext.storage().getFile(objectURI);
+        return new BufferedReader(new InputStreamReader(is));
+    }
+
+    @Builder
+    @Getter
+    public static class Output implements io.kestra.core.models.tasks.Output {
+        @Schema(
+            title = "The filtered file URI."
+        )
+        private final URI uri;
+
+        @Schema(
+            title = "The total number of records that was dropped by the task."
+        )
+        private final Long droppedRecordsTotal;
+    }
+
+    private static class PebbleExpressionPredicate implements ThrowingFunction<String, Boolean, Exception> {
+
+        protected static final ObjectMapper MAPPER = JacksonMapper.ofIon();
+        private final RunContext runContext;
+        private final String expression;
+
+        /** {@inheritDoc} */
+        @Override
+        public Boolean apply(String data) throws Exception {
+            try {
+                String rendered = extract(MAPPER.readTree(data));
+                return rendered == null ? null : Boolean.parseBoolean(rendered.trim());
```

Comment:
We usually coalesce boolean for convenience, see  io.kestra.core.utils.TruthUtils and how it is used in the If task